### PR TITLE
Add User table to postgresql db.

### PR DESCRIPTION
**Pull request by sorsby**
- Create User table, need to run make migrate-db upgrade-db to get the new database schema working.
- Added methods to User model.
- Add tests for create_user command.

### CODE REVIEW

**Review by MikeSilverstone — CHANGES_REQUESTED** [review]

File: api/commands/create_user_command.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    Prompts for new username, email and password to create a new admin account.
+    """
+
+    def run(self):
+        username = prompt(
+            'Please enter a username (max length 20 chars)')
+        email = prompt(
+            'Please enter an email address (max length 120 chars)')
+        password = prompt_pass(
```
[the remark this answers is not in the document]
Comment (sorsby) [reply api/commands/create_user_command.py]: What does this mean? prompt_pass hides the password as you type it.

File: api/test_runner.py [hunk 2]
```diff
@@ -27,6 +28,8 @@
 suite.addTest(loader.loadTestsFromModule(test_rounds_endpoint))
 suite.addTest(loader.loadTestsFromModule(test_single_round_endpoint))
 suite.addTest(loader.loadTestsFromModule(test_single_result_endpoint))
+suite.addTest(loader.loadTestsFromModule(test_command_create_user))
+
 
 # initialize a runner, pass it your suite and run it
 # runner = unittest.TextTestRunner(verbosity=3)
```
[the remark this answers is not in the document]
Comment (sorsby) [reply api/test_runner.py]: ??????

File: api/tests/test_command_create_user.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import unittest
+from unittest.mock import MagicMock, Mock
```
Comment (glossopm) [inline api/tests/test_command_create_user.py]: Mock isn't used?

**Review by glossopm — CHANGES_REQUESTED** [review]

File: api/manage.py [hunk 4]
```diff
@@ -1,17 +1,22 @@
 import os
-from flask_script import Manager
+from flask_script import Manager, prompt
 from flask_migrate import Migrate, MigrateCommand
 
+from flask_bcrypt import Bcrypt
```
[the remark this answers is not in the document]
Comment (sorsby) [reply api/manage.py]: yeet

File: api/tests/test_command_create_user.py (hunk 3 above)
Comment (sorsby) [reply api/tests/test_command_create_user.py]: yeet

File: api/test_runner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeSilverstone) [reply api/test_runner.py]: '#runner = unittest.TextTestRunner(verosity=3)'

is just a commented out line of code, does it need to be there?

File: api/commands/create_user_command.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikeSilverstone) [reply api/commands/create_user_command.py]: Cool. 👍

**Review by glossopm — APPROVED** [review]

File: api/tests/test_command_create_user.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    @mock.patch('commands.create_user_command.CreateUserCommand.hash_password',
+                MagicMock(return_value='test'))
+    @mock.patch('commands.create_user_command.prompt', MagicMock(return_value='test'))
+    @mock.patch('commands.create_user_command.prompt_pass', MagicMock(return_value='test'))
+    @mock.patch('commands.create_user_command.db.session.add', MagicMock(return_value=None))
+    @mock.patch('commands.create_user_command.db.session.commit', MagicMock(return_value=None))
+    def test_command_create_user(self):
+        new_user = self.command.run()
+
+        expected_user = User(username='test', email='test', password='test')
```
[the remark this answers is not in the document]
Comment (sorsby) [reply api/tests/test_command_create_user.py]: I added another test to check the password hashing, have a look see.

File: api/commands/create_user_command.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class CreateUserCommand(Command):
+    """CLI command which prompts for new username, email and password and stores in the database.
+
+    Prompts for new username, email and password to create a new admin account.
+    """
+
+    def run(self):
+        username = prompt(
```
[the remark this answers is not in the document]
Comment (sorsby) [reply api/commands/create_user_command.py]: Self sabotage? depends who is using the commands and also it wasn't on the card.
This is essentially just hardcoding the users and doing it directly in the db wouldn't have any validation either.

**Review by MikeSilverstone — APPROVED** [review]

**Pull request merged**